scrapy/spiders/__init__.py

```diff
@@ -68,6 +68,11 @@ def _set_crawler(self, crawler):
 
     def start_requests(self):
         cls = self.__class__
+        if not self.start_urls and hasattr(self, 'start_url'):
+            raise AttributeError(
+                "Crawling could not start: 'start_urls' list not found "
```

Comment:
I know [the docs](https://docs.scrapy.org/en/1.8/topics/spiders.html#scrapy.spiders.Spider.start_urls) do mention a "list", but since it could technically be [any iterable](https://github.com/scrapy/scrapy/blob/1.8.0/scrapy/spiders/__init__.py#L69-L70) I would remove the "list" part. Maybe "the 'start_urls' attribute" is good enough.

PS: looking at this, I figured it would be good to remove the `make_requests_from_url` method, it's been deprecated for a long time now. I'll open a PR for that.